internal/grpc/services/gateway/storageprovider.go

```diff
@@ -307,6 +307,11 @@ func (s *svc) getHome(_ context.Context) string {
 
 func (s *svc) InitiateFileDownload(ctx context.Context, req *provider.InitiateFileDownloadRequest) (*gateway.InitiateFileDownloadResponse, error) {
 	log := appctx.GetLogger(ctx)
+
+	if utils.IsRelativeReference(req.Ref) {
```

Comment:
> > How do we distinguish a simple request from a spaces request when the only difference is that for a spaces request the first path segment is the space id? The simple handler cannot just always take the first path segment as the space id. That is why we added a new datatx 'protocol': The path in the URL schema carries a different meaning.
> 
> I get the use case but I feel that it should be transparent to the clients. If a GRPC client wants to use relative paths, it should not know about the `spaces` concept, but just keep using it as it does for absolute paths.
> 
> Maybe we can add a check in the simple protocol. If the first routing in the URL path is of the type `storage-ID!opaque-ID`, then we frame the request reference accordingly?

Unfortunately, then any path that starts with that pattern will be interpreted as a relative spaces request. What about: `Hey! Cool Pics/2021-08-09/yay I can inject relative path references now.png` 

The only other way to reuse the `simple` datatx would be to use a dedicated path segment in the datatx urls:
- `/simple/absolute/path/to/file.ext`
- `/spaces/storage-ID!opaque-ID/relative/path/to/file.ext`

but it would break existing urls  ... well ... they are recerated whenever making an initiate download/upload call, so it should not be a problem. Would that be ok for you?

Reply:
Delete and Move both call getPath, which in turn calls stat. Together, they also do the necessary combined and absolute path handling.